pkg/controllers/machinemigration/machine_migration_controller_test.go

```diff
@@ -0,0 +1,693 @@
+		capiCluster        *capiv1beta1.Cluster
+	)
+
+	BeforeEach(func() {
+		By("Setting up namespaces for the test")
+		migrationControllerNamespace = corev1resourcebuilder.Namespace().
+			WithGenerateName("machine-migration-controller-").Build()
+		Expect(k8sClient.Create(ctx, migrationControllerNamespace)).To(Succeed(), "migration controller namespace should be able to be created")
+
+		mapiNamespace = corev1resourcebuilder.Namespace().
+			WithGenerateName("openshift-machine-api-").Build()
+		Expect(k8sClient.Create(ctx, mapiNamespace)).To(Succeed(), "MAPI namespace should be able to be created")
+
+		capiNamespace = corev1resourcebuilder.Namespace().
+			WithGenerateName("openshift-cluster-api-").Build()
+		Expect(k8sClient.Create(ctx, capiNamespace)).To(Succeed(), "CAPI namespace should be able to be created")
+
+		mapiMachineBuilder = machinev1resourcebuilder.Machine().
+			WithNamespace(mapiNamespace.GetName()).
+			WithName("foo").
+			WithProviderSpecBuilder(machinev1resourcebuilder.AWSProviderSpec().WithLoadBalancers(nil))
+
+		infrastructureName := "cluster-foo"
+		capaClusterBuilder = capav1builder.AWSCluster().
+			WithNamespace(capiNamespace.GetName()).
+			WithName(infrastructureName)
+		Expect(k8sClient.Create(ctx, capaClusterBuilder.Build())).To(Succeed(), "CAPA cluster should be able to be created")
+
+		capiClusterBuilder = capiv1resourcebuilder.Cluster().
+			WithNamespace(capiNamespace.GetName()).
+			WithName(infrastructureName)
+		Expect(k8sClient.Create(ctx, capiClusterBuilder.Build())).To(Succeed(), "CAPI cluster should be able to be created")
+
+		capiCluster = &capiv1beta1.Cluster{}
+		Expect(k8sClient.Get(ctx, client.ObjectKey{Name: infrastructureName, Namespace: capiNamespace.GetName()}, capiCluster)).To(Succeed())
+
+		capaMachineBuilder = capav1builder.AWSMachine().
+			WithNamespace(capiNamespace.GetName()).
+			WithName("machine-template")
+
+		capaMachine = capaMachineBuilder.Build()
+
+		capaMachineRef := corev1.ObjectReference{
+			APIVersion: capaMachine.APIVersion,
+			Kind:       capaMachine.Kind,
+			Name:       capaMachine.GetName(),
+			Namespace:  capaMachine.GetNamespace(),
+		}
+
+		capiMachineBuilder = capiv1resourcebuilder.Machine().
+			WithNamespace(capiNamespace.GetName()).
+			WithInfrastructureRef(capaMachineRef).
+			WithName("foo").
+			WithClusterName(infrastructureName)
+
+		reconciler = &MachineMigrationReconciler{
+			Client:        k8sClient,
+			Scheme:        testEnv.Scheme,
+			CAPINamespace: capiNamespace.GetName(),
+			MAPINamespace: mapiNamespace.GetName(),
+		}
+
+		k = komega.New(k8sClient)
+	})
+
+	AfterEach(func() {
+		By("Cleaning up MAPI test resources")
+		testutils.CleanupResources(Default, ctx, cfg, k8sClient, mapiNamespace.GetName(),
+			&machinev1beta1.Machine{},
+			&machinev1beta1.Machine{},
+			&configv1.Infrastructure{},
+		)
+		testutils.CleanupResources(Default, ctx, cfg, k8sClient, capiNamespace.GetName(),
+			&capiv1beta1.Machine{},
+			&capiv1beta1.Machine{},
```

Comment:
Machine twice?

Reply:
Search/Replace gone wrong, one should be MachineSet